service/history/workflow/context.go

```diff
@@ -1054,6 +1055,22 @@ func (c *ContextImpl) forceTerminateWorkflow(
 	)
 }
 
+// CacheSize estimates the in-memory size of the object for cache limits. For proto objects, it uses proto.Size()
+// which returns the serialized size. Note: In-memory size will be slightly larger than the serialized size.
+func (c *ContextImpl) CacheSize() int {
+	if !c.config.HistoryCacheLimitSizeBased() {
```

Comment:
Does this mean changes to this flag can take effect without restart? How do we safely switch from count-based to size-based in that case? The flags for max size only take effect after restart I think?

Reply:
If we switch at runtime we'd need to iterate over all cache elements to figure at where we're at before admitting new elements.

I don't see that here

Reply:
We don't want to allow changing this at runtime. Cache size will only change after restart as @yycptt mentioned. Let me see how to allow changing this only at startup.